src/library_sockfs.js

```diff
@@ -280,6 +280,13 @@ mergeInto(LibraryManager.library, {
 
         function handleMessage(data) {
           assert(typeof data !== 'string' && data.byteLength !== undefined);  // must receive an ArrayBuffer
+
+          // An empty ArrayBuffer will emit a pseudo disconnect event
```

Comment:
Ok, I see, so it must be sent manually.

Then, next question, why is the correct behavior to return immediately? what would happen otherwise?